blazar/plugins/devices/device_plugin.py

```diff
@@ -70,6 +73,27 @@ def __init__(self):
         super(DevicePlugin, self).__init__()
         self.plugins = self._get_plugins()
         self.placement_client = placement.BlazarPlacementClient()
+        self._check_resource_providers()
+
+    def _check_resource_providers(self):
+        # check if there is a reservation provider for all resource providers
+        # in placement
+        parent_resource_providers = []
+        reservation_resource_providers = {}
+        for rp in self.placement_client.list_resource_providers():
+            if re.match('^blazar_(.*)$', rp['name']):
+                reservation_resource_providers[rp['parent_provider_uuid']] = rp
+            else:
+                parent_resource_providers.append(rp)
```

Comment:
It can be the case that there are other resource providers that are not parents of the reservation one. It is not safe to assume that just b/c it's not a Blazar RP, that it must be an eligible parent.

I also hit a bug where Blazar was auto-creating RPs for every Zun host, even the hosts not enrolled in Blazar. This prevents Blazar from enrolling them, because it errors out on the RP already existing (probably this should be fixed.) I think that the RP should only be created lazily as part of this check if we know the device has been enrolled. So, I think this logic should pull the list of devices, then try to find all the RPs for those devices, then see if there are any device RPs that do not have a Blazar RP. Those are the ones that should be lazy auto-created.